servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/HttpAttributesGetter.java

```diff
@@ -152,42 +191,55 @@ public String getServerAddress(final RequestInfo requestInfo) {
             // For the server address we prefer the unresolved address, if possible. If we don't have that we'll
             // fall back to the resolved address.
             HostAndPort effectiveHostAndPort = requestInfo.request().effectiveHostAndPort();
-            return effectiveHostAndPort != null ? effectiveHostAndPort.hostName() :
-                    ServiceTalkHttpAttributesGetter.getResolvedAddress(requestInfo);
+            if (effectiveHostAndPort != null) {
+                return effectiveHostAndPort.hostName();
+            }
+            ConnectionInfo connectionInfo = requestInfo.connectionInfo();
+            if (connectionInfo == null) {
+                return null;
+            }
+            SocketAddress address = connectionInfo.remoteAddress();
+            if (address instanceof InetSocketAddress) {
+                return ((InetSocketAddress) address).getHostString();
+            } else if (address instanceof DomainSocketAddress) {
+                return ((DomainSocketAddress) address).getPath();
```

Comment:
Don't mind if we have an explicit path for `DomainSocketAddress`, just a heads up that `DomainSocketAddress.toString()` returns path, which allows use simply to use `address.toString()` for this case

Reply:
I'd prefer to explicitly get the path just in case `.toString()` changes in the future: its meaning is less rigid.